### ISSUE TITLE
Invocation of struct local activities is broken.

### ISSUE THREAD

**Issue by mfateev** [temporalio/sdk-go#424]
## Expected Behavior
```go
	var a *Activities

	var greetResult string
	err := workflow.ExecuteLocalActivity(ctx, a.GetGreeting).Get(ctx, &greetResult)
```
should work the same way it works for normal activities.

## Actual Behavior
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x18d07a4]

goroutine 70 [running]:
github.com/temporalio/samples-go/greetings.(*Activities).GetGreeting(...)
        /Users/maxim/temporal/temporal-go-samples/greetings/activities.go:17
reflect.Value.call(0x19454c0, 0xc000165910, 0x13, 0x1a72c0b, 0x4, 0x0, 0x0, 0x0, 0x203000, 0x203000, ...)
        /usr/local/go/src/reflect/value.go:476 +0x8c7
reflect.Value.Call(0x19454c0, 0xc000165910, 0x13, 0x0, 0x0, 0x0, 0x100f5f0, 0xc000588790, 0x10)
        /usr/local/go/src/reflect/value.go:337 +0xb9
go.temporal.io/sdk/internal.(*activityExecutor).executeWithActualArgsWithoutParseResult(0xc0005c2720, 0x1b801c0, 0xc000598ae0, 0x0, 0x0, 0x0, 0xc0005c09c0, 0x1b801c0, 0xc000598ae0)
        /Users/maxim/go/pkg/mod/go.temporal.io/sdk@v1.6.0/internal/internal_worker.go:830 +0x24a
go.temporal.io/sdk/internal.(*activityExecutor).ExecuteWithActualArgs(0xc0005c2720, 0x1b801c0, 0xc000598ae0, 0x0, 0x0, 0x0, 0x20a66c0, 0x202c7a9, 0xb)
        /Users/maxim/go/pkg/mod/go.temporal.io/sdk@v1.6.0/internal/internal_worker.go:804 +0x70
go.temporal.io/sdk/internal.(*localActivityTaskHandler).executeLocalActivityTask.func2(0x1b801c0, 0xc000598ae0, 0xc0004340e0, 0xc0005aaf00, 0xc0005c2720, 0xc00058c0d0, 0xc000588770, 0x1b88180, 0xc0005a2600, 0x2540be400, ...)
        /Users/maxim/go/pkg/mod/go.temporal.io/sdk@v1.6.0/internal/internal_task_pollers.go:563 +0x185
created by go.temporal.io/sdk/internal.(*localActivityTaskHandler).executeLocalActivityTask
        /Users/maxim/go/pkg/mod/go.temporal.io/sdk@v1.6.0/internal/internal_task_pollers.go:559 +0xa45

Process finished with the exit code 2
```

## Steps to Reproduce the Problem

Take the greetings sample and change it to use local activities instead of the normal ones.

**Comment by Sushisource** [comment]
Looks like https://github.com/temporalio/sdk-go/issues/405

**Comment by Sushisource** [comment]
@mfateev I don't think this can be fixed without a pretty big change in how local activities work. The problem seems to be if you pass a "method value" like `a.GetGreeting` there is no way to use reflection or anything else to access the receiver pointer, so we can't check that it's nil and assign it to a default (which still strikes me as a weird thing to do anyway).

We could instead look up the function name, and see if the activity struct has been registered and call it that way, but the problem is since we only know that a function was passed in, and we can't know about the receiver, the function looks valid and like we should try to call it rather than look it up in the registry, at which point it's too late.

So, I think we should just advise people to either initialize structs if they call this way (easiest by far), or to not do this at all, and instead call by string.

Technically we could also allow calling by "method expression", like `(*Activities).GetGreeting` but... not sure anyone cares.

**Comment by Sushisource** [comment]
Should be fixed by #436
